scripts/run-with-tarball.sh

```diff
@@ -0,0 +1,74 @@
+#! /usr/bin/env bash
+
+# Starts an rpc server and runs a tarball against it using rpc-client
+# Environment variables:
+#   DEFINITION: kore file to load (optional, otherwise from tarball)
+#   LLVM_LIB: default llvm library to load (required)
+#   BOOSTER: booster home directory (only relevant if not setting the following)
+#   SERVER: kore-rpc server path (default: from build dir in BOOSTER)
+#   CLIENT: rpc client path (default: from build dir in BOOSTER)
+#   LOG_DIR: path to place logs, defaults to current directory
+
+tarball=${1?"Tarball argument missing"}
+shift
+
+tarname=$(basename $tarball)
+
+booster=${BOOSTER:-$(realpath $(dirname $0)/..)}
+server=${SERVER:-$booster/.build/booster/bin/kore-rpc-booster}
+client=${CLIENT:-$booster/.build/booster/bin/kore-rpc-client}
+log_dir=${LOG_DIR:-.}
+
+if [ -z "$DEFINITION" ]; then
+    # unpack definition file from tarball, into tmp dir
+    mkdir -p tmp
+    tar xf $tarball -O definition.kore > tmp/definition.kore
+    kore=tmp/definition.kore
+else
+    kore=$DEFINITION
+fi
+
+if [ ! -s $kore ]; then
+    echo "Definition file $kore missing/empty"
+    exit 1
+fi
+
+
+# find out module name from definition file
+MODULE=$(grep -o -e "^module [A-Z0-9-]*" $kore | tail -1 | sed -e "s/module //")
+
+# hard-wire llvm backend, use surplus arguments from the command line as server options
+if [ -f ./${tarname}-interpreter.so ]; then
+    lib=./${tarname}-interpreter.so
+else
+    lib=$(realpath ${LLVM_LIB})
+fi
+if [ ! -f "$lib" ]; then
+    echo "A path to an LLVM backend library must be provided in LLVM_LIB"
+    exit 2
+fi
```

Comment:
I ended up extending my script today to call `llvm-kompile` and `llvm-kompile-matching`  so I could run `kontrol` tarballs better....
We would have to somehow pass in the `PLUGIN_DIR`  variable pointing to the nix store from the other script.

```suggestion
# build llvm backend unless provided
if [ -z "${LLVM_LIB}" ]; then
    mkdir -p tmp
    tar xf $tarball -O llvm_definition/definition.kore > tmp/llvm-definition.kore
    if [ ! -d "${PLUGIN_DIR}" ]; then
        echo "Either LLVM_LIB or PLUGIN_DIR must be provided"
        exit 2
    fi
    #generate matching data
    (cd tmp && mkdir -p dt && llvm-kompile-matching llvm-definition.kore qbaL ./dt 1/2)
    # find library dependencies and source files
    for lib in libff libcryptopp libsecp256k1; do
        LIBFILE=$(find ${PLUGIN_DIR} -name "${lib}.a" | head -1)
        [ -z "$LIBFILE" ] && (echo "[Error] Unable to locate ${lib}.a"; exit 1)
        PLUGIN_LIBS+="$LIBFILE "
        PLUGIN_INCLUDE+="-I$(dirname $LIBFILE)/../include "
    done
    PLUGIN_CPP="${PLUGIN_DIR}/include/plugin-c/blake2.cpp ${PLUGIN_DIR}/include/plugin-c/crypto.cpp ${PLUGIN_DIR}/include/plugin-c/plugin_util.cpp"

    # kompile llvm-definition to interpreter
    case "$OSTYPE" in
        linux*)
            LPROCPS="-lprocps"
            LIBSUFFIX="so"
            ;;
        *)
            LPROCPS=""
            LIBSUFFIX="dylib"
            ;;
    esac
    llvm-kompile tmp/llvm-definition.kore tmp/dt c -- \
             -fPIC -std=c++17 -o tmp/interpreter \
             $PLUGIN_LIBS $PLUGIN_INCLUDE $PLUGIN_CPP \
             -lcrypto -lssl $LPROCPS
    lib=tmp/interpreter.$LIBSUFFIX
else
    lib=$(realpath ${LLVM_LIB})
fi
if [ ! -f "$lib" ]; then
    echo "Problem with LLVM backend library at $lib"
    exit 2
fi
```